keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ClusterSpec.kt

```diff
@@ -0,0 +1,76 @@
+package com.netflix.spinnaker.keel.api.ec2
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.netflix.spinnaker.keel.api.Monikered
+import com.netflix.spinnaker.keel.api.ResourceSpec
+import com.netflix.spinnaker.keel.model.Moniker
+
+fun ClusterSpec.resolve(): Set<ServerGroupSpec> {
+  return locations.regions.map {
+    ServerGroupSpec(
+      moniker = moniker,
+      location = Location(
+        locations.accountName,
+        it.region,
+        it.subnet,
+        it.availabilityZones
+      ),
+      launchConfiguration = LaunchConfigurationSpec(
+        imageProvider = imageProvider,
+        instanceType = checkNotNull(overrides[it.region]?.launchConfiguration?.instanceType ?: defaults.launchConfiguration.instanceType),
+        ebsOptimized = checkNotNull(overrides[it.region]?.launchConfiguration?.ebsOptimized ?: defaults.launchConfiguration.ebsOptimized),
+        iamRole = checkNotNull(overrides[it.region]?.launchConfiguration?.iamRole ?: defaults.launchConfiguration.iamRole),
+        keyPair = checkNotNull(overrides[it.region]?.launchConfiguration?.keyPair ?: defaults.launchConfiguration.keyPair),
+        instanceMonitoring = overrides[it.region]?.launchConfiguration?.instanceMonitoring ?: defaults.launchConfiguration.instanceMonitoring ?: false,
+        ramdiskId = overrides[it.region]?.launchConfiguration?.ramdiskId ?: defaults.launchConfiguration.ramdiskId
+      ),
+      capacity = overrides[it.region]?.capacity ?: defaults.capacity ?: Capacity(1, 1, 1),
+      // TODO: need to be able to merge these rather than treating them as blocks
+      dependencies = overrides[it.region]?.dependencies ?: defaults.dependencies ?: Dependencies(),
+      health = overrides[it.region]?.health ?: defaults.health ?: Health(),
+      scaling = overrides[it.region]?.scaling ?: defaults.scaling ?: Scaling(),
+      tags = overrides[it.region]?.tags ?: defaults.tags ?: emptyMap()
+    )
+  }
+    .toSet()
+}
+
+data class ClusterSpec(
+  override val moniker: Moniker,
+  val imageProvider: ImageProvider,
+  val locations: ClusterLocations,
```

Comment:
Newbie question: this is `locations`, plural, but not a list, and then `ClusterLocations` below is mapped to a single account. Are all server groups assumed to be in the same account in a cluster, or was this meant to be a list?

Reply:
all server groups in a cluster need to be in the same account